init.lua

```diff
@@ -236,6 +241,34 @@ minetest.register_chatcommand("warp", {
 	end
 })
 
+local function prepare_dropdown(x,y,w,h,curr_dest)
+	local dd = string.format("dropdown[%f,%f;%f,%f;ddwarp;", x, y, w, h)
+	local sel = 0
+	for idx, warp in ipairs(warps) do
+		local warpname = warp.name
+		dd = dd .. warpname .. ","
```

Comment:
This leaves a tailing comma which likely results in an empty dropdown choice. You can use tables + `table.concat` to avoid that:

```Lua
local entries = {}
for k, v n ipairs(table_to_loop) do
	table.insert(entries, v)
end
return table.concat(entries, `,`) -- outputs  "value1,value2,value3" without tailing comma
```

Reply:
In this case we actually want an empty dropdown option because having nothing selected in the dropdown means we want to use the "future destination" field on the formspec, which enables warpstones to be set to places that don't exist yet. If that doesn't sound like a feature that makes sense, let me know and I'll remove it.

Reply:
As a placeholder that's fine. I didn't think of the "non-configured" state.